app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -1325,4 +1335,6 @@ private void updateProgressInfo(@NonNull final StreamInfo info) {
                     animateView(detailPositionView, false, 500);
                 });
     }
+
+    private static class RestrictedVideoException extends Exception { }
```

Comment:
Why create a new exception type to just use it to transfer information to a method? Why not just creating another method to handle restricted video infos?

Reply:
Because I was having some issue when calling `showError` directly, but it seems to work fine now.